File: java/BillsApi/src/main/java/com/apilangarena/javabillsapi/application/bills/create/BillWriteRepository.java
package com.apilangarena.javabillsapi.application.bills.create;

import com.apilangarena.javabillsapi.domain.bills.NewBill;

public interface BillWriteRepository {
    boolean existsByBillNumber(String billNumber);

    CreateBillResult create(NewBill newBill);
}

File: kotlin/BillsApi/src/main/kotlin/com/apilangarena/billsapi/application/bills/create/BillWriteRepository.kt
package com.apilangarena.billsapi.application.bills.create

import com.apilangarena.billsapi.domain.bills.NewBill

interface BillWriteRepository {
    fun existsByBillNumber(billNumber: String): Boolean
    fun create(newBill: NewBill): CreateBillResult
}
